### ISSUE TITLE
Upload is slow?

### ISSUE THREAD

**Issue by dsmit** [openstreetcam/ios#18]
When uploading to Mapillary I easily reach 4MB/s. However Openstreetview is between a few 100kb/sec to max 2MB/s with many upload stalls.

The uploading of my daily drives takes hours and need to be planned :(

**Comment by alexandrui-telenav** [comment]
Hi @dsmit, right now we're working on removing the stalls which should improve things.
We do have a plan to improve overall upload speeds, but to be honest we have some cost restrictions. This can easily be solved by using AWS but this will mean lots and lots of money. We have in mind a hybrid solution but I can't promise anything very soon.
In the meantime, I'll let you know when the stalls should stop, I'm looking forward to your feedback on that!

**Comment by hoylen** [comment]
This may be related, but I'm unable to upload due to "restarts" rather than "stalls".

By "restart" I mean it seems to upload for a few minutes (reaching about 6% of 263MB) and then goes back to 0% and starts again. I let it run for 30 minutes and it just still keeps restarting from the beginning, so it never finishes uploading.

P.S. It would be good if there is a way to upload individual tracks instead of just "upload all".

Version 1.4.1(2) for iOS.

**Comment by trailblazr** [comment]
Looks like I have the exact same issue with Version 1.4.3 on the iOS client. I stopped the endless-loop of uploading after a while because it obviously did more of a DDoS for the servers. **This might be a severe issue you will get on the backend (clients DDoS'ing you guys).**

**Comment by bogdans-telenav** [comment]
@hoylen as @trailblazr sad the issue has generated on the server side because of the security updated on the server side this happened around version 1.4.1 
To fix this you can log out and log in again with your OSM account.

**Comment by hoylen** [comment]
Thanks. The previous update seemed to have fixed the "restarts" I was experiencing (the one before 1.4.3(1), the one that came out on the App Store around 15 September 2016). I've managed to upload all my tracks, though one of them seems to be still "processing" after about 5 days.

Although I have now got version 1.4.3(2) (which appeared in the App Store around 24 September 2016), I will try logging out and back in again, just in case that helps.
